### PR TITLE
Parse description of (--data) flag cli

### PR DESCRIPTION
**Pull request by DylanVerstraete**
Fixes issue #490

### CODE REVIEW

File: pkg/cli/flag.go [hunk 1]
```diff
@@ -219,6 +223,50 @@ func EncodingTypeFlagDescription(mask EncodingType) string {
 	return "enum flag to define how to encode the output, options: " + strings.Join(options, "|")
 }
 
+// ArbitraryDataFlagVar defines a []modules.NetAddress flag with specified name and usage string.
```
Comment (GlenDC) [inline pkg/cli/flag.go]: Copy-pasting is OK, but do make sure the content is up-to-date, in this case, update comments

File: pkg/cli/flag.go [hunk 2]
```diff
@@ -219,6 +223,50 @@ func EncodingTypeFlagDescription(mask EncodingType) string {
...
+// The argument s points to a []modules.NetAddress variable in which to store the validated values of the flags.
+// The value of each argument will not try to be separated by comma, each value has to be defined as a separate flag (using the same name).
+func ArbitraryDataFlagVar(f *pflag.FlagSet, s *[]byte, name string, usage string) {
+	f.Var(&ArbitraryDataFlag{arbitraryData: s}, name, usage)
+}
+
+// ArbitraryDataFlagVarP defines a []modules.NetAddress flag with specified name, shorthand and usage string.
```
Comment (GlenDC) [inline pkg/cli/flag.go]: Copy-pasting is OK, but do make sure the content is up-to-date, in this case, update comments

File: pkg/cli/flag.go [hunk 3]
```diff
@@ -219,6 +223,50 @@ func EncodingTypeFlagDescription(mask EncodingType) string {
...
+// String implements pflag.Value.String
+func (flag *ArbitraryDataFlag) String() string {
+	byteArray := *flag.arbitraryData
+	if len(byteArray) == 0 {
+		return ""
+	}
+
+	s := strconv.Quote(string(byteArray[:]))
+
+	if s == "" {
```
Comment (GlenDC) [inline pkg/cli/flag.go]: this can never happen, as you already check `len(byteArray)`

File: pkg/cli/flag.go [hunk 4]
```diff
@@ -219,6 +223,50 @@ func EncodingTypeFlagDescription(mask EncodingType) string {
...
+		return ""
+	}
+	return strings.Trim(s, "\"")
```
Comment (GlenDC) [inline pkg/cli/flag.go]: Please check docs, to make sure what quotes it uses, is it always `"`, or can it also be another one? If so, you might need to check which one is used and trim that one.

File: pkg/cli/flag_test.go [hunk 5]
```diff
@@ -302,6 +303,44 @@ func testPanic(t *testing.T, label string, f func()) {
...
+			[]byte{0, 1, 2},
+		}, {
+			"2",
+			`\x00\x05\x13`,
+			[]byte{0, 5, 19},
+		}, {
+			"3",
+			`\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00`,
+			[]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+		},
```
Comment (GlenDC) [inline pkg/cli/flag_test.go]: Also check edge cases, in your case, the empty string. Plus also check regular strings, e.g. `"Hello, World!"`, as well as a string such as a unicode string escaped AND non-escaped.

File: pkg/client/walletcmd.go [hunk 6]
```diff
@@ -296,6 +291,12 @@ func createWalletCmd(cli *CommandLineClient) *WalletCommand {
 		&walletCmd.walletLoadSeedCfg.Seed,
 		"seed", "", "define the seed to be loaded as a flag instead of the STDIN")
 
+	// custom arbitrarydata flag
+	clipkg.ArbitraryDataFlagVar(sendCoinsCmd.Flags(), &walletCmd.sendCoinsCfg.Data,
+		"data", "optional arbitrary data (or description) to attach to transaction")
+	clipkg.ArbitraryDataFlagVar(sendCoinsCmd.Flags(), &walletCmd.sendBlockStakesCfg.Data,
```
Comment (GlenDC) [inline pkg/client/walletcmd.go]: copy-paste error

File: pkg/cli/flag.go (hunk 4 above)
Comment (DylanVerstraete) [reply pkg/cli/flag.go]: https://golang.org/src/strconv/quote.go?s=2985:3012#L104

`Quote returns a double-quoted Go string literal`

**Comment by GlenDC** [comment]
LGTM

**Pull request merged**